source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -306,6 +306,11 @@ bool SslSocketInfo::peerCertificatePresented() const {
   return cert != nullptr;
 }
 
+bool SslSocketInfo::localCertificatePresented() const {
+  bssl::UniquePtr<X509> cert(SSL_get_certificate(ssl_.get()));
```

Comment:
IIRC SSL_get_certificate doesn't increment the refcount of X509 so this shouldn't be UniquePtr, ASAN will catch if that's the case likely.